plugins/Jobs.cpp

```diff
@@ -1139,10 +1139,10 @@ bool BedrockPlugin_Jobs::processCommand(SQLite& db, BedrockCommand& command) {
         }
         const string& state = result[0][0];
 
-        // Make sure we're failing a job that's actually running
-        if (state != "RUNNING") {
-            SINFO("Trying to fail job#" << request["jobID"] << ", but isn't RUNNING (" << state << ")");
-            STHROW("405 Can only fail RUNNING jobs");
+        // Make sure we're failing a job that's actually running or running with a retryAfter
+        if (state != "RUNNING" && state != "RUNQUEUED") {
```

Comment:
@flodnv shouldn't we check timestamp here to identify if its a currently active RUNQUEUED job to ensure we don't fail a job that RUNQUEUED but not active?

Reply:
My opinion is that it's fine not to check it since we don't do it for `CancelJob` either.

Reply:
@chiragsalian What's a non-active RUNQUEUED job? RUNQUEUED is effectively the same as RUNNING, except with more features.

Reply:
He means that a RUNQUEUED job may be RUNNING or it may be QUEUED. Therefore, I think he is asking if it's ok to fail a RUNQUEUED job that may be just QUEUED waiting to be run, since we don't know if it's actually running or not.

Reply:
Yup what rocio said, RUNQUEUED is basically a job determined as RUNNING or QUEUED depending on its nextRun time. Hence the question being is it alright that we can fail a RUNQUEUED job thats effectively the same state as QUEUED i.e., a job that can be retried via getJob.

Reply:
Oh ok I see what you mean. 

> is it alright that we can fail a RUNQUEUED job

Yes, it is already past the "raw" QUEUED state, it is running, if you want to fail it, you can ; just like you can finish it or cancel it. If you don't do anything, it'll run again, and again, and again. In some situations you don't want to finish nor cancel it, so we need to allow for failing it.